qrenderdoc/Windows/ConstantBufferPreviewer.cpp

```diff
@@ -340,7 +346,7 @@ void ConstantBufferPreviewer::updateLabels()
                       .arg(IsD3D(pipeType) ? lit("CB") : lit("UBO"))
                       .arg(m_slot);
 
-  if(m_Ctx.CurPipelineState().SupportsResourceArrays())
+  if(m_Ctx.CurPipelineState().IsConstantBufferArray(m_stage, m_slot))
```

Comment:
I think instead of this specific function (which then contradicts `SupportsResourceArrays()`) we should instead leave this alone and return `true` from `SupportsResourceArrays()` for D3D12.

D3D12 has this ugly situation where it both does and doesn't properly support resource arrays which makes this messy unfortunately, but I think we have to pick an interpretation and stick to it - I think your approach of considering arrays 'real' is better so we should follow that and then fake it if/when necessary, which we already kind of do.

Reply:
My thinking here was that having a way to tell if a particular constant buffer is an array is useful, as only then should the title have the brackets to reduce some confusion. But I suppose that could have drawbacks too when switching between events and these panes stay around. Thoughts?

Reply:
Yeh having it always visible is consistent with vulkan and like you say it means it's more consistent when changing between different events where the viewer is implicitly for array element 0 but then it disappears. Especially if you e.g. open two viewers for array element 0 and array element 1 then switch to an event where that binding is non-arrayed. Only one viewer will be valid (the one for array element 0) but they'll have the same title.

Also it's a bit of a nit, but the intention of the common `PipeState` is to abstract around code that does a slightly different thing for each API. In this case the implementation can be done API agnostically - look up the binding in the relevant stage's bindpoint mapping and check if the `arraySize` is greater than 1.